client/boot/common.js

```diff
@@ -88,6 +88,15 @@ const loggedOutMiddleware = currentUser => {
 	);
 
 	page( '*', ( context, next ) => {
+		if ( '/' === context.pathname && config.isEnabled( 'devdocs/redirect-loggedout-homepage' ) ) {
```

Comment:
Unfortunately, I can't think of a better solution for this at this point. If we have multiple cases where:

- There is a section that users should only be able to access while logged in, and
- Users that visit this section while logged out should be redirected to a specific, non-login page,

then we might want to come up with a general solution for this. I don't think this is the case now, so this change seems reasonable to me.